waku/README.md

```diff
@@ -0,0 +1,310 @@
+    - [Packet Format and Usage](#packet-format-and-usage)
+    - [Whisper Envelope](#whisper-envelope)
+    - [Contents of Data Field of the Message (Optional)](#contents-of-data-field-of-the-message-optional)
+    - [Payload Encryption](#payload-encryption)
+- [Light node](#light-node)
+- [Rationale](#rationale)
+- [Backwards Compatibility](#backwards-compatibility)
+    - [WakuWhisper bridging](#wakuwhisper-bridging)
+- [Forwards Compatibility](#forwards-compatibility)
+- [Security Considerations](#security-considerations)
+- [Implementation](#implementation)
+- [Changelog](#changelog)
+    - [Differences between shh/6 waku/0](#differences-between-shh6-waku0)
+- [Copyright](#copyright)
+
+## Abstract
+
+This specification describes the format of Waku messages within the ÐΞVp2p Wire Protocol. This spec substitutes [EIP- 627](https://eips.ethereum.org/EIPS/eip-627). Waku is a fork of the original Whisper protocol that enables better scalability and offline messaging, at the cost of some metadata protection guarantees. It does this through (a) light node support (b) mailserver (c) waku mode and (d) basic rate limiting.
+
+## Motivation
+
+<!-- This should be the motivation of Waku rather than the motivation of this spec: aka to enable scalability etc. -->
+
+It is necessary to specify the standard for Waku messages in order to ensure forward compatibility of different Waku clients, backwards compatibility with Whisper clients, as well as to allow multiple implementations of Waku.
+
+## Specification
+
+### Use of RLPx transport protocol
+
+All Waku messages are sent as devp2p [RLPx transport protocol, version 5](https://github.com/ethereum/devp2p/blob/master/rlpx.md) packets. These packets MUST be RLP-encoded arrays of data containing two objects: packet code followed by another object (whose type depends on the packet code). 
+
+Waku nodes that do not support a particular packet code MUST ignore the packet without generating any error.
+
+Waku is a RLPx subprotocol called `waku` with version `0`. The version number corresponds to the major version in the header spec.
+
+### Packet Codes
+
+<!-- Packet code not clear -->
+
+The message codes reserved for Waku protocol: 0 - 127.
+
+<!-- Is this a MUST or rather a SHOULD -->
+
+Messages with unknown codes MUST be ignored, for forward compatibility of future versions.
+
+The Waku sub-protocol MUST support the following packet codes:
+
+| Name                       | Int Value |
+|----------------------------|-----------|
+| Status                     |     0     |
+| Messages                   |     1     |
+| PoW Requirement            |     2     |
+| Bloom Filter               |     3     |
+
+
+The following message codes are optional, but they are reserved for specific purpose.
+
+| Name                       | Int Value |
+|----------------------------|-----------|
+| P2P Request                |    126    |
+| P2P Message                |    127    |
+
+### Packet Format and Usage
```

Comment:
I dislike the way all these packets are documented, I feel like there should be a more concise way to do this.

Reply:
ABNF, also see https://github.com/vacp2p/specs/issues/43